xdsl/transforms/stencil_shape_minimize.py

```diff
@@ -42,6 +50,24 @@ def match_and_rewrite(self, op: func.FuncOp, rewriter: PatternRewriter, /):
             op.update_function_type()
 
 
+@dataclass
+class RestrictStoreOp(RewritePattern):
+    restrict: tuple[int, ...] | None = None
```

Comment:
This class might potentially always have a `restrict`, since it can't do much otherwise.

Reply:
that makes sense, I'll remove the default None value. thanks!